### ISSUE TITLE
Failure to invoke object primitive data type wrappers in default invocation resolver

### ISSUE THREAD

**Issue by terminalsin** [LLVM-but-worse/maple-ir#1]
**Stack trace**

`2020-11-27 21:12:49 INFO  Boot:304 - Running ConstantExpressionEvaluatorPass
Exception in thread "main" java.lang.RuntimeException: java.lang.NoSuchMethodError: java/lang/Double.java/lang/Double()D
        at org.mapleir.Boot.run(Boot.java:213)
        at org.mapleir.Boot.main(Boot.java:109)
Caused by: java.lang.NoSuchMethodError: java/lang/Double.java/lang/Double()D
        at org.mapleir.DefaultInvocationResolver.resolve(DefaultInvocationResolver.java:557)
        at org.mapleir.DefaultInvocationResolver.resolveVirtualCalls(DefaultInvocationResolver.java:623)
        at org.mapleir.ir.code.expr.invoke.VirtualInvocationExpr.resolveVirtualInvocation(VirtualInvocationExpr.java:93)
        at org.mapleir.ir.code.expr.invoke.DynamicInvocationExpr.resolveTargets(DynamicInvocationExpr.java:162)
        at org.mapleir.deob.interproc.IRCallTracer.traceInvocation(IRCallTracer.java:45)
        at org.mapleir.deob.interproc.IRCallTracer.traceImpl(IRCallTracer.java:38)
        at org.mapleir.deob.interproc.CallTracer.trace(CallTracer.java:37)
        at org.mapleir.deob.interproc.IPAnalysis.create(IPAnalysis.java:37)
        at org.mapleir.deob.interproc.IPAnalysis.create(IPAnalysis.java:30)
        at org.mapleir.deob.passes.constparam.ConstantExpressionEvaluatorPass.accept(ConstantExpressionEvaluatorPass.java:58)
        at org.mapleir.deob.PassGroup.accept(PassGroup.java:81)
        at org.mapleir.Boot.run(Boot.java:210)
        ... 1 more
`

**Root of the issue:**
Not entirely sure. Pretty sure it's due to nesting and unnesting of java primitives into their object forms and vice versa. Here's my sample used:
https://drive.google.com/file/d/17nKBJQab5dw7EHB9twYh74rMld5ZdLgu/view?usp=sharing

**Comment by rcx** [comment]
Hi, could you send the driver code you’re using to run maple-ir (especially
the list of passes)?

Thanks, and sorry for the delay.

**Comment by terminalsin** [comment]
Currently running the following:
https://github.com/terminalsin/maple-ir/blob/memes/production-ready/org.mapleir.main/src/main/java/org/mapleir/Boot.java

**Comment by rcx** [comment]
That’s unexpected... Boot.java doesn’t call for
ConstantExpressionEvaluatorPass. In any case, if this pass is enabled, try
disabling it. The interprocedural passes are old and unmaintained.

**Comment by terminalsin** [comment]
Understandable. Cheers for your time and many thanks for the creation of this tool. Ironically, I'm using it as of now as an experimental obfuscation API. Definitely one of the most straight forward and easy to interact with SSA bytecode API.

**Comment by rcx** [comment]
LOL Actually same here. But I didn’t expect other people would be able to
use it since I didn’t document it properly like I should have.

**Comment by rcx** [comment]
Also if you want an easier driver code to work with, check CleanBoot
instead of Boot

**Comment by terminalsin** [comment]
Had a look at it and still prefer the Boot class. I don't even use a boot class anylonger since I implemented my own boot in the obfuscation project.

And yes, shame on you for not detailing more the process of CFG manipulation using the API. There's still a lot of things I've got to learn. There's a bunch of examples nonetheless with the passes. I should be able to do fine.

Nonetheless, kudos once again for this. Will be dropping some PRs if I encounter any other bugs which are related to the IR itself. :)
